Bring back webkit calendar picker indicator for datalists

nkdas91 opened the pull request:
Fixes #35386 for the time being.

nkdas91 committed:
Bring back webkit calendar picker indicator for datalists

ffoodd commented on scss/_reboot.scss:
```diff
@@ -443,13 +443,6 @@ select {
-// See https://stackoverflow.com/a/54997118
-
-[list]::-webkit-calendar-picker-indicator {
-  display: none;
```
We may improve this by setting `opacity: 0` instead of `display: none`. This way we'd still remove the Chrome arrow with working `datalist`.

FWIW the weird arrow is a `list-style: disclosure-open inside` UA style, could be replaced I guess.

nkdas91 replied:
Yes, ```opacity: 0``` works. 

But on input types ```date```, ```time```, ```month```, etc, the datalist won't show up unless clicked on the indicator. So, if we hide it, the user won't know if he/she has to click on the extreme right of the field.

I have created a CodePen demo: https://codepen.io/nkdas91/pen/MWEXzvL where we can use ```opacity: 0``` for text type inputs and can replace the indicator with Bootstrap Icons if we want to.

Please advise. I'll make changes accordingly. Thanks!

ffoodd replied:
That's a really neat idea. Thanks!

I'd even go with a dropdown-like icon for text inputs, what do you think?

Another consideration is whether we should make it a default or not, since it'd be kinda heavy. We might revisit this when bundlewatch run.

ffoodd reviewed (changes requested):
I'd better keep the arrow removal, but I agree `datalist` need to work!
Thanks for the PR.

nkdas91 committed:
Replace webkit datalist indicator with Bootstrap Icons

nkdas91 commented:
Bundlewatch Failed.

I think

```
[list]:not([type="date"]):not([type="datetime-local"]):not([type="month"]):not([type="week"]):not([type="time"])::-webkit-calendar-picker-indicator {
 opacity: 0;
}
```

should be good to go.

The default datalist indicators for ```date``` and ```time``` are good enough in Chrome 96. Not sure about other versions.

ffoodd commented:
Using icons is kinda heavy indeed, so we'd better go with natives. Thanks for your patience!

Customizing icons could be something to document as a snippet, in #33627: cross-linking here for now :)

nkdas91 committed:
Use default datalist icons on date or time inputs in Chrome
Remove the dropdown arrow from text type inputs

nkdas91 commented:
Alright, I reverted the code to use native datalist icons on date or time inputs and removed the dropdown icon on text type inputs. 
Used ```display: none``` instead of ```opacity: 0``` because with ```opacity: 0``` the indicator was still taking up space which looked like padding.
Thanks!

ffoodd reviewed (approved):
LGTM! Thanks :heart:

mdo reviewed (approved):
It's a super gnarly selector, but I can see the value in restoring this so let's give it a go.

mdo committed:
Bundlewatch
